src/server.c

```diff
@@ -3968,6 +3968,17 @@ int processCommand(client *c) {
         }
     }
 
+    if (!server.cluster_enabled &&
+        server.masterhost &&
+        !mustObeyClient(c) &&
+        server.repl_replica_redirect_rw &&
+        (is_write_command ||
+         (is_read_command && !(c->flags & CLIENT_READONLY)))) {
+        addReplyErrorSds(c,sdscatprintf(sdsempty(), "-MOVED -1 %s:%d",
+                                                    server.masterhost, server.masterport));
```

Comment:
Presumably we need to also cover the "announced" port for K8s type distributions, where this port is not the same port the end user would see. I guess announced IP/hostnames are also problems.

Reply:
good point. but for that we need a new config, right?

Reply:
you mean `master-announce-ip` and `master-announce-port`?

Reply:
Yeah, something like that. Not sure how I feel about adding two new configurations that I would like to deprecate though. Ideally we would re-use the `cluster-announce-*` type configs. This relates to how I want to make sure we are aligned long term.

Reply:
For tls, we'd need `master-announce-tls-port` too then. Many new configs. Or maybe only these two: `redirect host:port` and `redirect-tls host:port`, i.e. redirect to any node, regardless of whether it's our master or not. Too flexible?

Reply:
I prefer @zuiderkwast 's idea, the `redirect` and `redirect-tls` has a clear scope, only affect redirect.

If we support `master-announce-*`, seems we need apply the announce ip:port to `info` `role` and other commands' reply. But this could cause some internal management services to become unavailable, since the internal services and user businesses possibly belong to different network, and the management may rely on the real ip:port in `info` and `role` etc.